src/sinks/aws_sns/integration_tests.rs

```diff
@@ -0,0 +1,114 @@
+#![cfg(all(test, feature = "aws-sns-integration-tests"))]
+
+use std::collections::HashMap;
+
+use aws_sdk_sns::{Client as SnsClient, Region};
+use codecs::TextSerializerConfig;
+use tokio::time::{sleep, Duration};
+
+use super::{config::SnsSinkConfig, sink::SnsSink};
+use crate::{
+    aws::{create_client, AwsAuthentication, RegionOrEndpoint},
+    common::sns::SnsClientBuilder,
+    config::ProxyConfig,
+    sinks::VectorSink,
+    test_util::{
+        components::{run_and_assert_sink_compliance, AWS_SINK_TAGS},
+        random_lines_with_stream, random_string,
+    },
+};
+
+fn sns_address() -> String {
+    std::env::var("SNS_ADDRESS").unwrap_or_else(|_| "http://localhost:4566".into())
+}
+
+async fn create_test_client() -> SnsClient {
+    let auth = AwsAuthentication::test_auth();
+
+    let endpoint = sns_address();
+    let proxy = ProxyConfig::default();
+    create_client::<SnsClientBuilder>(
+        &auth,
+        Some(Region::new("localstack")),
+        Some(endpoint),
+        &proxy,
+        &None,
+        true,
+    )
+    .await
+    .unwrap()
+}
+
+#[tokio::test]
+async fn sns_send_message_batch() {
+    let topic_name = gen_topic_name();
+    let topic_arn = ensure_topic(topic_name.clone()).await;
+
+    let client = create_test_client().await;
+
+    let config = SnsSinkConfig {
+        topic_arn: topic_arn.clone(),
+        region: RegionOrEndpoint::with_both("local", sns_address().as_str()),
+        encoding: TextSerializerConfig::default().into(),
+        message_group_id: None,
+        message_deduplication_id: None,
+        request: Default::default(),
+        tls: Default::default(),
+        assume_role: None,
+        auth: Default::default(),
+        acknowledgements: Default::default(),
+    };
+
+    config.clone().healthcheck(client.clone()).await.unwrap();
+
+    let sink = SnsSink::new(config, client.clone()).unwrap();
+    let sink = VectorSink::from_event_streamsink(sink);
+
+    let (mut input_lines, events) = random_lines_with_stream(100, 10, None);
+    run_and_assert_sink_compliance(sink, events, &AWS_SINK_TAGS).await;
+
+    sleep(Duration::from_secs(1)).await;
+
+    // Somewhat lost how to receive the published message
```

Comment:
seems the simplest option to test the roundtrip would be 
1. Creating a sqs queue which subscribes to the topic
2. Receive the message from the SQS queue